### PR TITLE
Del section

### PR DESCRIPTION
**Pull request by jmunzer**

### CODE REVIEW

**Review by jmunzer — COMMENTED** [review]
@timhodson, @lordtatty - would you mind reviewing & 'hopefully approving' this PR please?

Many thanks,
Joe

File: del_section/src/comp.php [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+//***********READ**DATA******************
+
+$file_handle = fopen($uploadfile, "rb");
+
+while (!feof($file_handle) )  {
+
+	$line_of_text = fgets($file_handle);
+	$parts = explode(" ", $line_of_text);
+	$barc = trim($parts[0]);
+	$item_lookup = 'https://rl.talis.com/3/' . $shortCode . '/draft_sections/' . $barc . '?include=list';
```
Comment (timhodson) [inline del_section/src/comp.php:116]:
```suggestion
	$section_lookup = 'https://rl.talis.com/3/' . $shortCode . '/draft_sections/' . $barc . '?include=list';
```

File: del_section/src/comp.php [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+	//************GRAB**LIST**DETAILS*************
+
+		$ch4 = curl_init();
+
+		curl_setopt($ch4, CURLOPT_URL, $item_lookup);
```
Comment (timhodson) [inline del_section/src/comp.php:122]:
```suggestion
		curl_setopt($ch4, CURLOPT_URL, $section_lookup);
```

File: del_section/src/comp.php [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+		
+		fwrite($myfile, $title . "\t");
+		fwrite($myfile, $assoc_listid . "\t");
+		fwrite($myfile, $title . "\t");
+		fwrite($myfile, $barc . "\t");
+
+	if ($shouldWritetoLive == "true") {
+
+	//**************DELETE_SECTION***************
+	$patch_url = 'https://rl.talis.com/3/' . $shortCode . '/draft_sections/' . $barc;
```
Comment (timhodson) [inline del_section/src/comp.php:159]:
```suggestion
	$section_to_delete = 'https://rl.talis.com/3/' . $shortCode . '/draft_sections/' . $barc;
```

File: del_section/src/comp.php [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+						"list_etag": "' . $etag . '",
+						"list_id": "' . $assoc_listid . '"
+					}
+				}';
+
+	//**************POST_THE_DELETE************
+
+	$ch2 = curl_init();
+
+	curl_setopt($ch2, CURLOPT_URL, $patch_url);
```
Comment (timhodson) [inline del_section/src/comp.php:172]:
```suggestion
	curl_setopt($ch2, CURLOPT_URL, $section_to_delete);
```

File: del_section/src/comp.php [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+//***********READ**DATA******************
+
+$file_handle = fopen($uploadfile, "rb");
+
+while (!feof($file_handle) )  {
+
+	$line_of_text = fgets($file_handle);
+	$parts = explode(" ", $line_of_text);
+	$barc = trim($parts[0]);
```
Comment (timhodson) [inline del_section/src/comp.php:115]: Not sure what `$barc` is? If it is the guid of the section, can we call it `section_guid`? 
This is seen a number of places throughout the script.

File: del_section/src/comp.php [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+	curl_close($ch5);
+
+	$etag2 = $output_json3->data->meta->list_etag;
+	echo "    Updated ETag: " . $etag2 . "</br>";
+	echo "    ---------------------------------------------------";
+	echo "</br>";
+
+	if ($shouldPublishLists === TRUE) {
+		//**************PUBLISH**LIST***************
+		$patch_url2 = 'https://rl.talis.com/3/' . $shortCode . '/draft_lists/' . $assoc_listid . '/publish_actions';
```
Comment (timhodson) [inline del_section/src/comp.php:225]: This needs to be using the bulk publish lists method.
This endpoint is the public publish queue.

Incidentally. using the bulk publish queue means you don't need to get an etag again and so will mean you don't need to do the extra API request to get the list just above this.

**Review by timhodson — CHANGES_REQUESTED** [review]
The main comment is that we should be using the bulk publish queue rather than the publish actions for each list. Using publish actions will impact on the public queue.

Also a few minor variable name change suggestions.

File: del_section/src/comp.php (hunk 5 above)
Comment (jmunzer) [reply del_section/src/comp.php:115]: mm, $barc originally stood for barcode.. taken from an earlier php tool I wrote for our lms a long time ago.. 

It's just the variable name for the uuid pulled out of the input file. I agree this should be changed to something more relevant! section_guid is ideal :))

File: del_section/src/comp.php (hunk 6 above)
Comment (jmunzer) [reply del_section/src/comp.php]: endpoint & request body updated.

**Pull request merged**